src/components/SingleTrip.js

```diff
@@ -1,33 +1,20 @@
 import "../styles/SingleTrip.css";
-import { getSingleTripFromLocalStorage } from "../services/tripsStorage";
-import { useEffect, useState } from "react";
 import { useParams, Link } from "react-router-dom";
 import TripDetails from "./TripDetails";
 import Carousel from "./Carousel";
 import Weather from "./Weather";
 import GoBackButton from "./GoBackButton";
+import useSingleTrip from "../hooks/useSingleTrip";
 
 function SingleTrip() {
-  const [singleTrip, setSingleTrip] = useState({});
+  const [singleTrip] = useSingleTrip({});
   const { id } = useParams();
-  const [firstYear, setFirstYear] = useState("");
-  const [firstMonth, setFirstMonth] = useState("");
-  const [firstDay, setFirstDay] = useState("");
-  const [secondYear, setSecondYear] = useState("");
-  const [secondMonth, setSecondMonth] = useState("");
-  const [secondDay, setSecondDay] = useState("");
-
-  useEffect(() => {
-    const myTrip = getSingleTripFromLocalStorage(id);
-    setSingleTrip(myTrip);
-
-    setFirstYear(myTrip.dates[0].year);
-    setFirstMonth(myTrip.dates[0].month.name);
-    setFirstDay(myTrip.dates[0].day);
-    setSecondYear(myTrip.dates[1].year);
-    setSecondMonth(myTrip.dates[1].month.name);
-    setSecondDay(myTrip.dates[1].day);
-  }, [id]);
+  const [firstYear] = useSingleTrip("");
+  const [firstMonth] = useSingleTrip("");
+  const [firstDay] = useSingleTrip("");
+  const [secondYear] = useSingleTrip("");
+  const [secondMonth] = useSingleTrip("");
+  const [secondDay] = useSingleTrip("");
```

Comment:
i think here you can say (see below) (if you are using all of them) if not, i think you have to be careful because you are returning an array and then you have to pay attention to the order of the items inside the array.
If you returned an object instead of an array you can deconstruct it without paying attention to the order.
```suggestion
  const [singleTrip,
    setSingleTrip,
    firstYear,
    setFirstYear,
    firstMonth,
    setFirstMonth,
    firstDay,
    setFirstDay,
    secondYear,
    setSecondYear,
    secondMonth,
    setSecondMonth,
    secondDay,
    setSecondDay] = useSingleTrip()
```